lib/src/binding_coap/coap_extensions.dart

```diff
@@ -137,17 +163,24 @@ extension ResponseExtension on CoapResponse {
     }
   }
 
-  String get _contentType {
-    // FIXME: Replace once new CoAP library version has been released.
-    if (contentFormat == CoapMediaType.undefined) {
-      return 'application/json';
-    }
-
-    return CoapMediaType.name(contentFormat);
-  }
+  String get _contentType =>
+      contentFormat?.contentType.toString() ?? 'application/json';
 
   /// Extract the [Content] of this [CoapResponse].
   Content get content {
     return Content(_contentType, _payloadStream);
   }
+
+  /// Validates the payload and returns a serialized ACE creation hint if
+  /// successful.
+  List<int>? get creationHint {
+    final responsePayload = payload;
+    if (responsePayload != null) {
+      // TODO(JKRhb): Should the payload simply be returned here instead?
+      return AuthServerRequestCreationHint.fromSerialized(
+        responsePayload.toList(),
+      ).serialize();
+    }
+    return null;
+  }
```

Comment:
As stated in the TODO, would it make sense to directly pass the serialized creation hint here instead? Or is it useful to first ensure that the payload can be deserialized as a creation hint?

Reply:
This depends on how the rest of the library is structured — i.e., is it usually the responsibility of the caller to check whether any data is valid, or does `dart_wot` offer to do this?
Personally, I tend towards validating it here, so that we conform closely to the ACE-OAuth standard.

Reply:
> This depends on how the rest of the library is structured — i.e., is it usually the responsibility of the caller to check whether any data is valid, or does dart_wot offer to do this?

In general, I think library users should be able to expect that all returned data is valid. So I think validating here does make sense :)

>  Personally, I tend towards validating it here, so that we conform closely to the ACE-OAuth standard.

👍